Assets/Scripts/Simulation/SimulationBase.cs

```diff
@@ -238,11 +238,13 @@ public void UpdateDebugInfo()
             }
         }
 
+        private bool _hasWrittenStatistics;
         public virtual void OnSimulationFinished()
         {
-            if (GlobalSettings.ShouldWriteCsvResults)
+            if (GlobalSettings.ShouldWriteCsvResults && !_hasWrittenStatistics)
```

Comment:
HasWrittenStatistics is fucked. OnSimulationFinished should only be called once per simulation.